python/setup.py

```diff
@@ -88,111 +123,103 @@ def build_extension(self, ext):
             f"-DCMAKE_CONFIGURATION_TYPES:STRING={build_type}",
         ]
         cmake_args.extend(CMAKE_ARGS)
+        cmake_args.extend(environment_cmake_args)
 
         env = os.environ.copy()
-        Path(self.build_temp).mkdir(exist_ok=True)
 
         try:
             subprocess.run(
-                cmake_args, 
+                cmake_args,
                 cwd=self.build_temp,
                 env=env,
                 capture_output=True,
                 check=True)
         except subprocess.CalledProcessError as e:
-            print(e.stdout.decode("utf-8"))
+            print("stdout:", e.stdout.decode("utf-8"), "\n\nstderr:",
+                  e.stderr.decode("utf-8"))
             raise
         try:
             subprocess.run(
-                ["cmake", "--build", ".", "--config", build_type],
+                ["cmake", "--build", ".", "--config", build_type, "--verbose"],
                 cwd=self.build_temp,
                 env=env,
                 capture_output=True,
                 check=True
             )
         except subprocess.CalledProcessError as e:
-            print(e.stdout.decode("utf-8"))
+            print("stdout:", e.stdout.decode("utf-8"), "\n\nstderr:",
+                  e.stderr.decode("utf-8"))
             raise
 
         cmake_bdist_dir = Path(self.build_temp) / Path(ext.python_package_dir)
         dir_util.copy_tree(
             str(cmake_bdist_dir / ext.name), str(Path(output_dir) / ext.name)
         )
 
-
-requirements = [
+extra_requirements = [
```

Comment:
Make hard requirements of these, until the "index"-idea has been implemented (or something similar).